src/Superstructure/ESMFMod/interface/ESMF_Init_C.F90

```diff
@@ -59,14 +59,26 @@ subroutine f_esmf_frameworkinitialize(lang, configFileName, &
 
    end subroutine f_esmf_frameworkinitialize
 
-   subroutine f_esmf_frameworkfinalize(rc)
+   subroutine f_esmf_frameworkfinalize(rc, keepMpi)
        use ESMF_CompMod
        use ESMF_InitMod
+       use ESMF_UtilTypesMod
+       use, intrinsic :: iso_c_binding
 
        implicit none
 
        integer :: rc
+       logical(kind=c_bool), intent(in), optional :: keepMpi
+       type(ESMF_End_Flag) :: endflag
 
-       call ESMF_Finalize(rc=rc)
+       if(present(keepMpi)) then
```

Comment:
In the absent `else` case of `present(keepMpi)`, what's the default value of `endflag` that's getting passed down?

Reply:
It would've been an uninitialized value if I remember Fortran semantics correctly, but take a look at [7249651](https://github.com/esmf-org/esmf/pull/234/commits/72496513fa15a0d867dd15d0be73dab5f0ad3e3a), which now sets `endflag` to default to `ESMF_END_NORMAL` and changes it if `keepMpi` is present and true.